src/test_utils.rs

```diff
@@ -39,6 +41,49 @@ pub const AVATAR_900x900_BYTES: &[u8] = include_bytes!("../test-data/image/avata
 static CONTEXT_NAMES: Lazy<std::sync::RwLock<BTreeMap<u32, String>>> =
     Lazy::new(|| std::sync::RwLock::new(BTreeMap::new()));
 
+pub struct AcManager {
+    log_tx: Sender<Event>,
+    _log_sink: LogSink,
+    accounts: RefCell<Vec<Rc<TestContext>>>,
```

Comment:
I had somehow had in mind that you can't use `RefCell` and `Rc` with async, so I was suprised this works... But it compiles, so it should be fine.

Reply:
RefCell is !Send & !Sync so now AcManager can not be used from multiple threads or be moved to another thread.  just appears that so far this isn't needed because you do all the things in the thread of the test.

Reply:
I'd preferto go further here and don't use interior mutability at all.  If you want to be able to add an account to the AcManager you need to declare it `mut`.

Reply:
> I'd preferto go further here and don't use interior mutability at all.

Ah yes, when working on satisfying the borrow checker, I missed that we don't need this anymore... Removed it.

> so now AcManager can not be used from multiple threads or be moved to another thread

But I thought that at every `.await`, the task could be moved to a different thread?

Reply:
> But I thought that at every `.await`, the task could be moved to a different thread?

I think the runtime can only do this if your future is `Send`.  It's a bit hidden in async-std (tokio lets you explicitly build runtimes and has a single-threaded runtime) but there is a `spawn_local` to spawn `!Send` futures as well, because the normal `spawn` will possibly send you to other threads.

Reply:
Ok, thanks for explaining!